src/App.tsx

```diff
@@ -15,47 +16,113 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
+enum SortType {
+  NONE,
+  ALPHABET,
+  LENGTH,
+}
+
 export const App: React.FC = () => {
+  const [isReversed, setReverse] = useState(false);
+  const [sortOrder, setSortOrder] = useState(SortType.NONE);
+
+  const reset = () => {
+    setSortOrder(SortType.NONE);
+    setReverse(false);
+  };
+
+  const getReorderedGoods = (
+    goods: string[],
+    sortType: SortType,
+    reversed: boolean,
+  ) => {
+    const visibleGoods = [...goods];
+
+    visibleGoods.sort((good1, good2) => {
+      switch (sortType) {
+        case SortType.ALPHABET:
+          return good1.localeCompare(good2);
+
+        case SortType.LENGTH:
+          return good1.length - good2.length;
+
+        default:
+          return 0;
+      }
+    });
+
+    if (reversed) {
+      visibleGoods.reverse();
+    }
+
+    return visibleGoods;
+  };
+
+  const reorderedGoods = getReorderedGoods(
+    goodsFromServer,
+    sortOrder,
+    isReversed,
+  );
+
+  const isDefaultOrder = sortOrder === SortType.NONE
+  && !isReversed;
+
   return (
+
     <div className="section content">
       <div className="buttons">
         <button
           type="button"
-          className="button is-info is-light"
+          className={cn('button is-info', {
+            'is-light': sortOrder !== SortType.ALPHABET,
+          })}
+          onClick={() => setSortOrder(SortType.ALPHABET)}
         >
           Sort alphabetically
         </button>
 
         <button
           type="button"
-          className="button is-success is-light"
+          className={cn('button is-success', {
+            'is-light': sortOrder !== SortType.LENGTH,
+          })}
+          onClick={() => setSortOrder(SortType.LENGTH)}
         >
           Sort by length
         </button>
 
         <button
           type="button"
-          className="button is-warning is-light"
+          className={cn('button is-warning', {
+            'is-light': !isReversed,
+          })}
+          onClick={() => setReverse(!isReversed)}
         >
           Reverse
         </button>
 
-        <button
-          type="button"
-          className="button is-danger is-light"
-        >
-          Reset
-        </button>
+        {(!isDefaultOrder) && (
+          <button
+            type="button"
+            className={cn('button is-danger is-light')}
+            onClick={reset}
+          >
+            Reset
+          </button>
+        )}
       </div>
 
       <ul>
         <ul>
-          <li data-cy="Good">Dumplings</li>
-          <li data-cy="Good">Carrot</li>
-          <li data-cy="Good">Eggs</li>
-          <li data-cy="Good">Ice cream</li>
-          <li data-cy="Good">Apple</li>
-          <li data-cy="Good">...</li>
+          {reorderedGoods
+            .map(good => (
+              <li
+                key={good}
+                data-cy="Good"
+              >
```

Comment:
Do you need likebreaks when you have only 2 props?
```suggestion
              <li key={good} data-cy="Good">
```